### PR TITLE
fix: temporary fix for vector issue #498

### PR DESCRIPTION
**Pull request by ikrommyd**
Temporary fix to bypass https://github.com/scikit-hep/vector/issues/498

### CODE REVIEW

File: src/egamma_tnp/nanoaod_efficiency.py [hunk 1]
```diff
@@ -512,6 +513,13 @@ def __repr__(self):
         return f"PhotonTagNProbeFromNanoAOD(Filters: {self.filters}, Number of files: {n_of_files})"
 
     def find_probes(self, events, cut_and_count, mass_range, vars):
+        # TODO: remove this temporary fix when https://github.com/scikit-hep/vector/issues/498 is resolved
+        photon_dict = {field: events.Photon[field] for field in events.Photon.fields} | {
+            "mass": dak.zeros_like(events.Photon.pt),
+            "charge": dak.zeros_like(events.Photon.pt),
+        }
+        events["Photon"] = dak.zip(photon_dict, with_name="Photon", behavior=candidate.behavior)
```
Comment (ikrommyd) [inline src/egamma_tnp/nanoaod_efficiency.py:521]: @Saransh-cpp could you let me me know what behavior should I specify here. Maybe `coffea.nanoevents.methods import candidate` and `behavior=nanoaod.behavior`. This is me not understanding well how behaviors work. I don't even know if this does anything in this case, I'm just trying to make sure I'm not missing anything to make the new photons behave like they should.

Comment (Saransh-cpp) [reply src/egamma_tnp/nanoaod_efficiency.py:521]: Hi! It should be `behavior=nanoaod.behavior`. `candidate.behavior` is included within `nanoaod.behavior`. coffea does not update the global awkward `behavior` (which is a good thing), which is why you need that extra `behavior=` arg.

Alternatively, to skip the `behavior` keyword, you could do something like -
```py
>>> ak.behavior.update(nanoaod.behavior)
>>> a = ak.zip({"x": [1], "y": [2], "z": [3], "m": [4]}, with_name="Photon")
>>> a
<PhotonArray [{x: 1, y: 2, z: 3, m: 4}] type='1 * Photon[x: int64, y: int64...'>
```

Makes me wonder if coffea should have something like [`vector.register_awkward`](https://github.com/scikit-hep/vector/blob/45bcb1a0fcc573d0326d5d1af0b6a2a4ad09a16b/src/vector/__init__.py#L177-L195) to make it easy for the users.

File: tests/dummy_tag_and_probe_nanoaod.py [hunk 2]
```diff
@@ -121,6 +122,14 @@ def tag_and_probe_electrons(events, is_id):
 
 
 def tag_and_probe_photons(events, start_from_diphotons, is_id):
+    # TODO: remove this temporary fix when https://github.com/scikit-hep/vector/issues/498 is resolved
+    photon_dict = {field: events.Photon[field] for field in events.Photon.fields} | {
+        "mass": ak.zeros_like(events.Photon.pt),
+        "charge": ak.zeros_like(events.Photon.pt),
+    }
+
+    events["Photon"] = ak.zip(photon_dict, with_name="Photon", behavior=candidate.behavior)
```
Comment (Saransh-cpp) [inline tests/dummy_tag_and_probe_nanoaod.py:131]: Similarly, should be `nanoaod.behavior`

**Comment by ikrommyd** [comment]
Thanks @Saransh-cpp :)

**Pull request merged**